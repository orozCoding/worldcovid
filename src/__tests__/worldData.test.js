import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import { worldData } from '../components/testsComponents/statsMock';
import WorldData from '../components/world/worldData';

describe('World Data', () => {
  it('renders', () => {
    let tree;
    act(() => {
      tree = renderer
        .create(
          <WorldData worldData={worldData} />,
        )
        .toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('displays date', () => {
    act(() => {
      render(<WorldData worldData={worldData} />);
    });
    expect(screen.getByText('2022-03-07')).toBeDefined();
  });

  it('displays stats', () => {
    act(() => {
      render(<WorldData worldData={worldData} />);
    });
    expect(screen.getByText('447,338,425')).toBeDefined();
  });
});
