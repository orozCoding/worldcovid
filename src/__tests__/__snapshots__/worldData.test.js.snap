// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`World Data renders 1`] = `null`;
